SharpPcap/LibPcap/PcapDevice.cs

```diff
@@ -586,6 +586,51 @@ public virtual void SendPacket(byte[] p, int size)
         /// <param name="size">The number of bytes to send</param>
         public abstract void SendPacket(ReadOnlySpan<byte> p);
 
+        #region Timestamp
+        public virtual TimestampResolution TimestampPrecision
+        {
+            get => (TimestampResolution)LibPcapSafeNativeMethods.pcap_get_tstamp_precision(PcapHandle);
```

Comment:
what happens if device not open?

Reply:
we should throw I think. I know its not recommended to throw from properties but we aren't the normal class since we interact with hardware. I'm ok if we break that rule vs. making a method.

Reply:
Let me rephrase: what happens when you do `LibPcapSafeNativeMethods.pcap_get_tstamp_precision(IntPtr.Zero)` ?

Reply:
I'd guess bad things occur, we should check and throw there.